pkg/storage/gcp/saver.go

```diff
@@ -0,0 +1,102 @@
+package gcp
+
+import (
+	"fmt"
+	"strings"
+
+	"cloud.google.com/go/storage"
+	"github.com/bketelsen/buffet"
+	"github.com/gobuffalo/buffalo"
+	"google.golang.org/api/option"
+)
+
+// Storage implements the (github.com/gomods/pkg/storage).Saver interface
+type Storage struct {
+	bucket *storage.BucketHandle
+}
+
+// New returns a new Storage instance
+// authenticated using the provided ClientOptions
+func New(ctx buffalo.Context, projectID, bucket string, cred option.ClientOption) (*Storage, error) {
+	client, err := storage.NewClient(ctx, cred)
+	if err != nil {
+		return nil, fmt.Errorf("could not create new client: %s", err)
+	}
+	// TODO: client should be closed on any error or when done
+	// should this be done after save is complete?
+
+	// The bucket MUST already exist
+	bkt := client.Bucket(bucket)
```

Comment:
any downside to just hard-coding bucket to `gomodules` or similar? maybe the caller could do it too, either way I can't think of a reason that two `Storage` instances should point to different buckets, you?

Reply:
not at all, I think I meant to change that.